pkg/kanctl/validate.go

```diff
@@ -55,17 +57,23 @@ func newValidateCommand() *cobra.Command {
 }
 
 func performValidation(cmd *cobra.Command, args []string) error {
+	var ctx context.Context
 	p, err := extractValidateParams(cmd, args)
 	if err != nil {
 		return err
 	}
 	cmd.SilenceUsage = true
-
+	ctx = cmd.Context()
+	if ctx == nil {
+		ctx = context.Background()
+	}
```

Comment:
Can you please check if calling `Execute()` sets the cmd.Context? If not, the ideal fix would be to change the `Execute()` with `ExecuteContext()` instead of handling it here. 
https://pkg.go.dev/github.com/spf13/cobra#Command.Context

Reply:
As per discussion, `Execute()` function by default creates `context.Background()`, hence we dont need to explicitly if the `ctx` is nil